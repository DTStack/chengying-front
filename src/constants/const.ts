// 部署状态
export const deployStatus = {
};

// 部署状态filter
export const deployStatusFilter = [
  {
    text: '部署成功',
    value: 'deployed',
  },
  {
    text: '部署失败',
    value: 'deploy fail',
  },
  {
    text: '部署中',
    value: 'deploying',
  },
  {
    text: '卸载失败',
    value: 'undeploy fail',
  },
  {
    text: '卸载中',
    value: 'undeploying',
  },
];

// 更新状态filter
export const updateStatusFilter = [
  {
    text: '更新成功',
    value: 'success',
  },
  {
    text: '更新失败',
    value: 'fail',
  },
  {
    text: '更新中',
    value: 'update',
  },
];

export const hostsInstallStatus = [
  { text: '管控安装成功', value: '管控安装成功' },
  { text: '管控安装失败', value: '管控安装失败' },
  { text: 'script安装成功', value: 'script安装成功' },
  { text: 'script安装失败', value: 'script安装失败' },
  { text: '主机初始化成功', value: '主机初始化成功' },
  { text: '主机初始化失败', value: '主机初始化失败' },
];

export const kubernetesInstallStatus = [
  { text: '管控安装失败', value: '管控安装失败', status: -1 },
  { text: '管控安装成功', value: '管控安装成功', status: 1 },
  { text: '主机初始化失败', value: '主机初始化失败', status: -3 },
  { text: '主机初始化成功', value: '主机初始化成功', status: 3 },
  { text: 'K8S DOCKER初始化失败', value: 'K8S DOCKER初始化失败', status: -5 },
  { text: 'K8S DOCKER初始化成功', value: 'K8S DOCKER初始化成功', status: 5 },
  { text: 'K8S NODE初始化失败', value: 'K8S NODE初始化失败', status: -6 },
  { text: 'K8S NODE初始化成功', value: 'K8S NODE初始化成功', status: 6 },
  { text: 'K8S NODE部署失败', value: 'K8S NODE部署失败', status: -7 },
  { text: 'K8S NODE部署成功', value: 'K8S NODE部署成功', status: 7 },
];

export const clusterTypeMap = {
  hosts: ['主机集群'],
  kubernetes: ['Kubernetes集群（自建集群）', 'Kubernetes集群（导入已有集群）'],
};

// k8s & hosts 主机初始化状态
export const hostStatusInfoMap = {
  hosts: {
    title:
      '主机安装会经过2个状态：管控安装-主机初始化，若某一步安装失败，请查看具体日志。',
    filters: hostsInstallStatus,
    finalStatus: 3,
  },
  kubernetes: {
    title:
      'Kubernetes集群主机安装会经过5个状态：管控安装-主机初始化-K8S Docker初始化-K8S Node初始化-K8S Node部署成功，若某一步安装失败，请查看具体日志',
    filters: kubernetesInstallStatus,
    finalStatus: 7,
  },
};

export const COOKIES = {
  NAMESPACE: 'em_current_k8s_namespace',
};
