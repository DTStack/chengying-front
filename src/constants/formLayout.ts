export const formItemCenterLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
  },
};

export const formItemBaseLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 16 },
  },
};
