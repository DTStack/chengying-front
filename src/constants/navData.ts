export const navData = [
  {
    title: '运维中心',
    icon: 'emicon-jiqunguanli',
    code: 'menu_ops_center',
    url: '/opscenter',
    level: 'first',
    routers: [
      '/opscenter/overview',
      '/opscenter/service',
      '/opscenter/hoststatus',
      'sub_menu_diagnose',
    ],
    children: [
      {
        title: '概览',
        icon: 'emicon-gailan',
        code: 'sub_menu_overview',
        url: '/opscenter/overview',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '服务',
        icon: 'emicon-fuwu',
        code: 'sub_menu_service',
        url: '/opscenter/service',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '主机',
        icon: 'emicon-zhuji',
        code: 'sub_menu_host',
        url: '/opscenter/hoststatus',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '诊断',
        icon: 'emicon-zhenduan',
        code: 'sub_menu_diagnose',
        url: '/opscenter/diagnosis',
        level: 'second',
        routers: [
          '/opscenter/diagnosis/log',
          '/opscenter/diagnosis/event',
          '/opscenter/diagnosis/config',
          '/opscenter/diagnosis/inspectionReport',
          '/opscenter/diagnosis/backup',
        ],
        children: [
          {
            title: '日志查看',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_diagnose_log',
            url: '/opscenter/diagnosis/log',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '事件诊断',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_diagnose_event',
            url: '/opscenter/diagnosis/event',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '配置变更',
            icon: 'emicon-peizhibiangeng',
            code: 'sub_menu_diagnose_config_change',
            url: '/opscenter/diagnosis/config',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '巡检报告',
            icon: 'emicon-xunjianbaogao',
            code: 'sub_menu_diagnose',
            url: '/opscenter/diagnosis/inspectionReport',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '备份管理',
            icon: 'emicon-beifenguanli',
            code: 'sub_menu_diagnose_backup',
            url: '/opscenter/diagnosis/backup',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
    ],
  },
  {
    title: '部署中心',
    icon: 'emicon-yingyongguanli',
    code: 'menu_deploy_center',
    url: '/deploycenter',
    level: 'first',
    routers: [
      '/deploycenter/cluster',
      '/deploycenter/appmanage',
      '/deploycenter/deployed',
    ],
    children: [
      {
        title: '集群管理',
        icon: 'emicon-jiqunguanli1',
        code: 'sub_menu_cluster_manage',
        url: '/deploycenter/cluster',
        level: 'second',
        routers: [
          '/deploycenter/cluster/list',
          '/deploycenter/cluster/create',
          '/deploycenter/cluster/create/edit',
          '/deploycenter/cluster/create/add',
          '/deploycenter/cluster/detail/index',
          '/deploycenter/cluster/detail/host',
          '/deploycenter/cluster/detail/imagestore',
          '/deploycenter/cluster/detail/namespace',
          '/deploycenter/cluster/detail/deployed',
          '/deploycenter/cluster/detail/history',
          '/deploycenter/cluster/detail/patchHistory',
          '/deploycenter/cluster/detail/echoList',
          '/deploycenter/cluster/detail/echoDetail',
        ],
        children: [
          {
            title: '集群列表',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_cluster_list',
            url: '/deploycenter/cluster/list',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '添加集群',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_cluster_edit',
            url: '/deploycenter/cluster/create',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
      {
        title: '组件管理',
        icon: 'emicon-zujianguanli',
        code: 'sub_menu_product_manage',
        url: '/deploycenter/appmanage',
        level: 'second',
        routers: [
          '/deploycenter/appmanage/products',
          '/deploycenter/appmanage/installs',
        ],
        children: [
          {
            title: '组件列表',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_product_list',
            url: '/deploycenter/appmanage/products',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '部署组件',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_product_deploy',
            url: '/deploycenter/appmanage/installs',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
      {
        title: '告警监控',
        icon: 'emicon-jiankonggaojing2',
        code: 'sub_menu_monitor',
        url: '/deploycenter/deployed',
        level: 'second',
        routers: [
          '/deploycenter/monitoring/dashboard',
          '/deploycenter/monitoring/dashdetail',
          '/deploycenter/monitoring/alert',
          '/deploycenter/monitoring/addAlert',
        ],
        children: [
          {
            title: '仪表盘',
            icon: 'emicon-rizhichakan',
            code: 'sub_menu_dashboard',
            url: '/deploycenter/monitoring/dashboard',
            level: 'three',
            routers: [],
            children: [],
          },
          {
            title: '告警',
            icon: 'emicon-shijianzhenduan',
            code: 'sub_menu_alert',
            url: '/deploycenter/monitoring/alert',
            level: 'three',
            routers: [],
            children: [],
          },
        ],
      },
    ],
  },
  {
    title: '平台管理',
    icon: 'emicon-yingyongguanli',
    code: 'menu_platform_manager',
    url: '/platform',
    level: 'first',
    routers: ['/platform/scriptManager', '/platform/backup', '/platform/clusterInspection'],
    children: [
      {
        title: '备份配置',
        icon: 'emicon-shijianzhenduan',
        code: 'sub_menu_configuration_backup_config',
        url: '/platform/backup',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '脚本管理',
        icon: 'emicon-yunweijiaobenpingtaibeifen',
        code: 'sub_menu_platform_manager_scriptManager',
        url: '/platform/scriptManager',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '集群巡检',
        icon: 'emicon-yunweijiaobenpingtaibeifen',
        code: 'sub_menu_platform_manager_cluster_inspection',
        url: '/platform/clusterInspection',
        level: 'second',
        routers: [],
        children: [],
      },
    ],
  },
  {
    title: '系统配置',
    icon: '',
    code: 'menu_system_configuration',
    url: '/systemconfig',
    level: 'first',
    routers: ['/systemconfig/security', '/systemconfig/globalConfig'],
    children: [
      {
        title: '平台安全',
        icon: 'emicon-jiankonggaojing2',
        code: 'sub_menu_configuration_platformsecurity_config',
        url: '/systemconfig/security',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '全局配置',
        icon: 'emicon-shijianzhenduan',
        code: 'sub_menu_configuration_platformsecurity_config_view',
        url: '/systemconfig/globalConfig',
        level: 'second',
        routers: [],
        children: [],
      },
    ],
  },
  {
    title: '用户管理',
    icon: '',
    code: 'menu_user_manage',
    url: '/usercenter',
    level: 'first',
    routers: [
      '/usercenter/members',
      '/usercenter/role',
      '/usercenter/role/view',
      '/usercenter/selfinfo',
      '/usercenter/audit',
    ],
    children: [
      {
        title: '成员管理',
        icon: 'emicon-chengyuanguanli',
        code: 'sub_menu_user_manage',
        url: '/usercenter/members',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '角色管理',
        icon: 'emicon-jiaoseguanli',
        code: 'sub_menu_role_manage',
        url: '/usercenter/role',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '个人信息',
        icon: 'emicon-gerenxinxi',
        code: 'sub_menu_user_info',
        url: '/usercenter/selfinfo',
        level: 'second',
        routers: [],
        children: [],
      },
      {
        title: '安全审计',
        icon: 'emicon-shenji',
        code: 'menu_security_audit',
        url: '/usercenter/audit',
        level: 'second',
        routers: [],
        children: [],
      },
    ],
  },
];

export const clusterNavData = [
  {
    title: '总览',
    icon: 'emicon-gailan',
    code: 'sub_menu_detail_overview',
    isShow: true,
    key: 'index',
  },
  {
    title: '主机资源',
    icon: 'emicon-fuwu',
    code: 'sub_menu_detail_host',
    isShow: true,
    key: 'host',
  },
  {
    title: '镜像仓库',
    icon: '',
    code: 'sub_menu_detail_image_store',
    isShow: true,
    key: 'imagestore',
  },
  {
    title: '命名空间',
    icon: '',
    code: 'sub_menu_detail_namespace',
    isShow: true,
    key: 'namespace',
  },
  {
    title: '已部署组件',
    icon: '',
    code: 'sub_menu_detail_deployed',
    isShow: true,
    key: 'deployed',
  },
  {
    title: '部署历史',
    icon: '',
    code: 'sub_menu_detail_history',
    isShow: true,
    key: 'history',
  },
  {
    title: '补丁包',
    icon: '',
    code: 'sub_menu_detail_patch',
    isShow: true,
    key: 'patchHistory',
  },
  {
    title: '集群命令',
    icon: '',
    isShow: true,
    key: 'echoList',
  },
  {
    title: '集群命令详情',
    icon: '',
    isShow: false,
    key: 'echoDetail',
  },
];
