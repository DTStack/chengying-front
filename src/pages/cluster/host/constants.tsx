import * as React from 'react';
import { Badge, Tag, Tooltip, Icon, message } from 'antd';
import ProgressBar from '@/components/progressBar';
import './style.scss';

export const columnGenerator = ({
  statusInfo,
  isKubernetes,
  isKubernetesCustom,
  authorityList,
  progressProp,
  hostGroups,
  handleRoleSet,
}) => {
  const roleAuth = authorityList.sub_menu_role_manage;
  return [
    {
      title: '节点',
      dataIndex: 'hostname',
      render: (text, record) => {
        return (
          <div>
            <p>{record.hostname}</p>
            <p className="text-gray">{record.ip}</p>
          </div>
        );
      },
    },
    {
      title: 'agent状态',
      dataIndex: 'is_running',
      filters: [
        { text: '运行中', value: 'true' },
        { text: '已停止', value: 'false' },
      ],
      filterMultiple: true,
      render: (text) => {
        return (
          <Badge
            color={text ? '#12BC6A' : '#FF5F5C'}
            text={text ? '运行中' : '已停止'}
          />
        );
      },
    },
    {
      title: (
        <span>
          主机初始化状态
          <Tooltip title={statusInfo.title}>
            <Icon type="info-circle" />
          </Tooltip>
        </span>
      ),
      dataIndex: 'status',
      width: '15%',
      filters: statusInfo.filters,
      filterMultiple: true,
      render: (text, record: any) => {
        const finalStatus = statusInfo.finalStatus;
        return (
          <React.Fragment>
            <Badge
              color={progressProp(record.status).color}
              text={progressProp(record.status).title || record.errorMsg}
            />
            {record.status !== finalStatus && (
              <Tooltip title={record.errorMsg}>
                &nbsp;
                <Icon type="info-circle" />
              </Tooltip>
            )}
          </React.Fragment>
        );
      },
    },
    isKubernetes
      ? {
          title: '角色',
          dataIndex: 'roles',
          width: '18%',
          filters: [
            { text: 'Control', value: 'Control' },
            { text: 'Worker', value: 'Worker' },
            { text: 'Etcd', value: 'Etcd' },
            { text: '全部', value: 'all' },
          ],
          filterMultiple: true,
          render: (text, record: any) => {
            const roles = [];
            for (const key in record.roles) {
              record.roles[key] && roles.push(key);
            }
            return (
              <React.Fragment>
                {roles.map((item) => (
                  <Tag className="c-cluster__tag" key={item}>
                    {item}
                  </Tag>
                ))}
              </React.Fragment>
            );
          },
        }
      : undefined,
    {
      title: '主机分组',
      dataIndex: 'group',
      filters: hostGroups,
      filterMultiple: true,
    },
    {
      title: 'CPU',
      dataIndex: 'cpu_usage_pct',
      sorter: true,
      render: (text, record: any) => {
        return (
          <ProgressBar
            now={record.cpu_core_used_display}
            total={record.cpu_core_size_display}
          />
        );
      },
    },
    isKubernetes
      ? {
          title: 'POD',
          dataIndex: 'pod_usage_pct',
          sorter: true,
          render: (text, record: any) => {
            return (
              <ProgressBar
                now={record.pod_used_display}
                total={record.pod_size_display}
              />
            );
          },
        }
      : {
          title: '磁盘',
          dataIndex: 'disk_usage_pct',
          sorter: true,
          render: (text, record: any) => {
            return (
              <ProgressBar
                pStyle={{ width: 140, marginLeft: '-28px' }}
                now={record.disk_used_display}
                total={record.disk_size_display}
              />
            );
          },
        },
    !isKubernetesCustom
      ? {
          title: '角色',
          dataIndex: 'role_list_display',
          ellipsis: true,
          render: (roleList = []) => {
            return roleList.length === 0 ? (
              '--'
            ) : (
              <Tooltip title={roleList.map((role) => role.role_name).join(',')}>
                <span>{roleList.map((role) => role.role_name).join(',')}</span>
              </Tooltip>
            );
          },
        }
      : undefined,
    !isKubernetesCustom
      ? {
          title: '操作',
          render: (value, record) => {
            record.role_list_display = record.role_list_display || [];
            const roleList = record.role_list_display.map(
              (item) => item.role_id
            );
            return (
              <span
                className="link-btn"
                onClick={() => {
                  if (roleAuth) {
                    handleRoleSet(record.sid, roleList);
                  } else {
                    message.error('权限不足，请联系管理员！');
                  }
                }}>
                编辑角色
              </span>
            );
          },
        }
      : undefined,
  ].filter((item) => item !== undefined);
};
