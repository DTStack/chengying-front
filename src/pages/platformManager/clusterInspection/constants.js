export const form_Title_info = [
    {
        "form_title": 'service_gc_statistics',
        "desc": "服务 Full GC 统计"
    },
    {
        "form_title": 'dir_usage_statistics',
        "desc": "/opt/dtstack/目录使用统计"
    },
    {
        "form_title": 'alarm_indicators',
        "desc": "告警指标列表"
    }
]

export const chartTable = [
    {
        'insert': 'HDFS 文件存储使用率',
        "form_title": 'hdfs_file_usage',
        "desc": "HDFS 文件存储使用率列表"
    },
    {
        'insert': 'CPU使用率趋势',
        "form_title": 'cpu_usage',
        "desc": "CPU使用率列表"
    },
    {
        'insert': '内存使用率趋势',
        "form_title": 'men_usage',
        "desc": "内存使用率列表 (最新)"
    },
    {
        'insert': 'swap 使用量趋势',
        "form_title": 'swap_usage',
        "desc": "swap使用列表 (最新)"
    },
    {
        'insert': '磁盘使用率趋势',
        "form_title": 'disk_usage',
        "desc": "磁盘使用率列表 (最新)"
    },
    {
        'insert': 'inode 使用率趋势',
        "form_title": 'inode_usage',
        "desc": "inode 使用率列表 (最新)"
    },
]
export const colorRange = [
    '#2FC25B',
    '#1890FF',
    '#6087EF',
    '#7460EF',
    '#AB60EF',
    '#E85EB3',
    '#EF5350',
    '#F18646',
    '#FFB310',
    '#8BDA63',
  ];

export const bigDataNodeList = [
    {
        "bigData_Node_title": 'NameNode内存(最新)',
        "name_node_memo": ""
    },
    {
        "bigData_Node_title": 'DataNode内存(最新)',
        "data_node_mem": ""
    },
    {
        "bigData_Node_title": 'DataNode存活数(最新)',
        "data_node_live_nums": ""
    },
    {
        "bigData_Node_title": 'DataNode死亡数(最新)',
        "data_node_dead_nums": ""
    },
    {
        "bigData_Node_title": 'HDFS文件数(最新)',
        "hdfs_file_num": ""
    }
]

