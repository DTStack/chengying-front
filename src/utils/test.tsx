import * as React from 'react';
import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';

// this is a handy function that I normally make available for all my tests
// that deal with connected components.
// you can provide initialState for the entire store that the ui is rendered with
export function renderWithRedux(
  Component,
  rootReducer,
  initialState: any = {}
) {
  const store: any = createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunkMiddleware)
  );
  const layout = <Provider store={store}>{Component}</Provider>;
  return {
    ...render(layout),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
