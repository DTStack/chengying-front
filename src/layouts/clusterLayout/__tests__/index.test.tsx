import * as React from 'react';
import ClusterLayout from '../index';
import reducer from '@/stores';
import { renderWithRedux } from '@/utils/test';
import { BrowserRouter } from 'react-router-dom';
import { clusterNavData } from '@/constants/navData';
import { fireEvent, RenderResult, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const defaultProps = {
  HeaderStore: {
    cur_product: {
      product_id: -1,
      product_name: '选择产品',
    },
    products: [],
    cur_parent_product: '选择产品', // 当前产品
    parentProducts: [],
    cur_parent_cluster: {
      id: 1,
      mode: 0,
      name: 'dtstack',
      type: 'hosts',
    },
    parentClusters: [],
  },
  location: {},
  history: {},
  match: {},
};

describe('test clusterlayout navbar', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = renderWithRedux(
      <BrowserRouter>
        <ClusterLayout {...defaultProps} />
      </BrowserRouter>,
      reducer,
      { HeaderStore: defaultProps.HeaderStore }
    );
  });

  afterEach(cleanup);

  test('test tabs render', () => {
    const tabsNav = wrapper.container.getElementsByClassName('ant-tabs-nav')[0];
    const tabs = tabsNav.getElementsByClassName('ant-tabs-tab');
    const tabpanes =
      wrapper.container.getElementsByClassName('ant-tabs-tabpane');
    const clusterTypeMaps = {
      hosts: {
        0: ['imagestore', 'namespace'],
      },
      kubernetes: {
        0: ['namespace', 'patchHistory', 'echoList'], // 自建
        1: ['index', 'host', 'patchHistory', 'echoList'], // 导入
      },
    };
    const {
      HeaderStore: {
        cur_parent_cluster: { mode, type },
      },
    } = defaultProps;
    const clusterMaps = clusterTypeMaps[type];
    // 过滤条件
    const filterConditions = clusterMaps[mode];
    // 过滤不同集群类型菜单
    const realClusterData = clusterNavData.filter(
      (nav) => nav.isShow && !filterConditions.includes(nav.key)
    );
    for (const i in tabs) {
      if (tabs[i] instanceof HTMLElement) {
        expect(tabs[i].textContent).toBe(realClusterData[i].title);
      }
    }
    expect(tabs.length).toBe(tabpanes.length);
  });

  test('test cluster render', () => {
    const a = wrapper.container.getElementsByClassName(
      'ant-dropdown-trigger'
    )[0];
    fireEvent.click(a);
    expect(a).toBeInTheDocument();
    const title = a.getElementsByClassName('title')[0];
    expect(title.textContent).toBe('dtstack');
  });

  test('test snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
