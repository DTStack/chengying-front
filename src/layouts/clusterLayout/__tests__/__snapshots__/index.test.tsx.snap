// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test clusterlayout navbar test snapshot 1`] = `
<DocumentFragment>
  <section
    class="container ant-layout"
  >
    <div
      class="container-header"
    >
      <div
        class="bread ant-breadcrumb"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            <a
              href="/deploycenter/cluster/list"
            >
              集群管理
            </a>
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            集群详情
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
      </div>
      <a
        class="ant-dropdown-link ant-dropdown-trigger"
      >
        <span>
          <span
            class="title"
          >
            dtstack
          </span>
           
          <i
            aria-label="icon: down"
            class="anticon anticon-down"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </a>
    </div>
    <div
      class="ant-tabs ant-tabs-top ant-tabs-line"
    >
      <div
        class="ant-tabs-bar ant-tabs-top-bar"
        role="tablist"
        tabindex="0"
      >
        <div
          class="ant-tabs-nav-container"
        >
          <span
            class="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
            unselectable="unselectable"
          >
            <span
              class="ant-tabs-tab-prev-icon"
            >
              <i
                aria-label="icon: left"
                class="anticon anticon-left ant-tabs-tab-prev-icon-target"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </i>
            </span>
          </span>
          <span
            class="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
            unselectable="unselectable"
          >
            <span
              class="ant-tabs-tab-next-icon"
            >
              <i
                aria-label="icon: right"
                class="anticon anticon-right ant-tabs-tab-next-icon-target"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                  />
                </svg>
              </i>
            </span>
          </span>
          <div
            class="ant-tabs-nav-wrap"
          >
            <div
              class="ant-tabs-nav-scroll"
            >
              <div
                class="ant-tabs-nav ant-tabs-nav-animated"
              >
                <div>
                  <div
                    aria-disabled="false"
                    aria-selected="true"
                    class="ant-tabs-tab-active ant-tabs-tab"
                    role="tab"
                  >
                    总览
                  </div>
                  <div
                    aria-disabled="false"
                    aria-selected="false"
                    class=" ant-tabs-tab"
                    role="tab"
                  >
                    主机资源
                  </div>
                  <div
                    aria-disabled="false"
                    aria-selected="false"
                    class=" ant-tabs-tab"
                    role="tab"
                  >
                    已部署组件
                  </div>
                  <div
                    aria-disabled="false"
                    aria-selected="false"
                    class=" ant-tabs-tab"
                    role="tab"
                  >
                    部署历史
                  </div>
                  <div
                    aria-disabled="false"
                    aria-selected="false"
                    class=" ant-tabs-tab"
                    role="tab"
                  >
                    补丁包
                  </div>
                  <div
                    aria-disabled="false"
                    aria-selected="false"
                    class=" ant-tabs-tab"
                    role="tab"
                  >
                    集群命令
                  </div>
                </div>
                <div
                  class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        role="presentation"
        style="width: 0px; height: 0px; overflow: hidden; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
        style="margin-left: 0%;"
      >
        <div
          aria-hidden="false"
          class="ant-tabs-tabpane ant-tabs-tabpane-active"
          role="tabpanel"
        >
          <div
            role="presentation"
            style="width: 0px; height: 0px; overflow: hidden; position: absolute;"
            tabindex="0"
          />
          <div
            role="presentation"
            style="width: 0px; height: 0px; overflow: hidden; position: absolute;"
            tabindex="0"
          />
        </div>
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
          role="tabpanel"
        />
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
          role="tabpanel"
        />
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
          role="tabpanel"
        />
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
          role="tabpanel"
        />
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
          role="tabpanel"
        />
      </div>
      <div
        role="presentation"
        style="width: 0px; height: 0px; overflow: hidden; position: absolute;"
        tabindex="0"
      />
    </div>
    <main
      class="ant-layout-content"
    />
  </section>
</DocumentFragment>
`;
